apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/logged-out-view/submit-buttons.js

```diff
@@ -0,0 +1,188 @@
+	}
+
+	if ( textNode ) {
+		fallbackTextColor = getComputedStyle( textNode ).color;
+	}
+
+	return {
+		fallbackBackgroundColor: backgroundColorValue || fallbackBackgroundColor,
+		fallbackTextColor: textColorValue || fallbackTextColor,
+	};
+} );
+
+/**
+ * Block edit function
+ *
+ * @typedef { import('@wordpress/block-editor').ColorPalette.Color } Color
+ * @property { string } class
+ * @property { string } color
+ * @property { string } name
+ * @property { string } slug
+ *
+ * @typedef { import('./').Attributes } Attributes
+ * @typedef {object} Props
+ * @property { Color } backgroundButtonColor
+ * @property { Color } textButtonColor
+ * @property { string } fallbackBackgroundColor
+ * @property { string } fallbackTextColor
+ * @property { Attributes } attributes
+ * @property { () => void } setBackgroundButtonColor
+ * @property { () => void } setTextButtonColor
+ * @property { (attributes: Partial<Attributes>) => void } setAttributes
+ *
+ * @param { Props } props
+ */
+function SubmitButtons( props ) {
+	useEffect( () => {
+		setButtonClasses();
+		setCustomButtonColors();
+	}, [ props.backgroundButtonColor, props.textButtonColor ] );
+
+	function setButtonClasses() {
+		const buttonClasses = getButtonClasses();
+		props.setAttributes( { buttonClasses } );
+	}
+
+	function setCustomButtonColors() {
+		const customTextButtonColor = getTextButtonColor();
+		const customBackgroundButtonColor = getBackgroundButtonColor();
+
+		if ( customTextButtonColor !== undefined ) {
+			props.setAttributes( { customTextButtonColor } );
+		}
+
+		if ( customBackgroundButtonColor !== undefined ) {
+			props.setAttributes( { customBackgroundButtonColor } );
+		}
+	}
+
+	function getTextButtonColor() {
+		return get( props.textButtonColor, 'color' );
+	}
+
+	function getBackgroundButtonColor() {
+		return get( props.backgroundButtonColor, 'color' );
+	}
+
+	function getButtonClasses() {
+		const { textButtonColor, backgroundButtonColor } = props;
+		const textClass = get( textButtonColor, 'class' );
+		const backgroundClass = get( backgroundButtonColor, 'class' );
+		return classnames( 'wp-block-button__link', {
+			'has-text-color': textButtonColor.color,
+			[ textClass ]: textClass,
+			'has-background': backgroundButtonColor.color,
+			[ backgroundClass ]: backgroundClass,
+		} );
+	}
+
+	const {
+		attributes,
+		setAttributes,
+		backgroundButtonColor,
+		textButtonColor,
+		setBackgroundButtonColor,
+		setTextButtonColor,
+		fallbackBackgroundColor,
+		fallbackTextColor,
+	} = props;
+
+	const backgroundColor = backgroundButtonColor.color || fallbackBackgroundColor;
+	const color = textButtonColor.color || fallbackTextColor;
+	const buttonStyle = { border: 'none', backgroundColor, color };
+	const buttonClasses = getButtonClasses();
+
+	return (
+		<div>
+			<div className="wp-block-button premium-content-logged-out-view-button">
+				<RichText
+					placeholder={ __( 'Add text…', 'full-site-editing' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **103** times:
`translate( 'Add text…' )` *ES Score: 9*